### PR TITLE
#1818 Implement data interface APIs for dynaflow HVDC

### LINKED ISSUE #1818: Requiring new data interface API for HVDC in dynaflow
rework API  for VSC converters

### PR DESCRIPTION
**Pull request by lecourtoisflo**
closes #1818

### CODE REVIEW

**Comment by lecourtoisflo** [comment]
The code smell doesn't need fixing

File: dynawo/sources/Modeler/DataInterface/DYNVscConverterInterface.h [hunk 1]
```diff
@@ -59,6 +60,17 @@ class VscConverterInterface : public ConverterInterface {
    * @return The maximum reactive power of the converter in Mvar (following iidm convention)
    */
   virtual double getQMax() = 0;
+
+  /**
+   * @brief Getter for the miinmum reactive power of the converter
+   * @return The miinmum reactive power of the converter in Mvar (following iidm convention)
```
Comment (marcochiaramello) [inline dynawo/sources/Modeler/DataInterface/DYNVscConverterInterface.h:66]: typo minimum

**Review by marcochiaramello — CHANGES_REQUESTED** [review]

File: dynawo/sources/Modeler/DataInterface/DYNVscConverterInterface.h (hunk 1 above)
Comment (lecourtoisflo) [reply dynawo/sources/Modeler/DataInterface/DYNVscConverterInterface.h:66]: OK

**Review by marcochiaramello — APPROVED** [review]

File: dynawo/sources/Modeler/DataInterface/DYNGeneratorInterface.h [hunk 2]
```diff
@@ -150,8 +130,7 @@ class GeneratorInterface : public ComponentInterface {
   virtual void exportStateVariablesUnitComponent() = 0;
 
   /**
-   * @brief Retrieve the list of reactive curve points, if any
-   * @returns list of reactive curve points
+   * @copydoc ReactiveCurvePointsInterface::getReactiveCurvesPoints() const
```
Comment (cossartque) [inline dynawo/sources/Modeler/DataInterface/DYNGeneratorInterface.h:133]: I would have called it getReactiveCurvePoints and not getReactiveCurvesPoints (no "s" at the end of Curve") to be consistent

File: dynawo/sources/Modeler/DataInterface/DYNReactiveCurvePointsInterface.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
+//
+// Copyright (c) 2021, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
```
Comment (cossartque) [inline dynawo/sources/Modeler/DataInterface/DYNReactiveCurvePointsInterface.h:10]: copyright is "open source suite of simulation tools for power systems" and not "open source time domain simulation tool for power systems" (nb: this should normally be the case everywhere in DFL)

File: dynawo/sources/Modeler/DataInterface/DYNReactiveCurvePointsInterface.h [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+#include <vector>
+
+namespace DYN {
+/// @brief Curve points data interface
+class ReactiveCurvePointsInterface {
+ public:
+  /**
+   * @brief Reactive curve point
+   *
+   * Represents a point extracted from network file
```
Comment (cossartque) [inline dynawo/sources/Modeler/DataInterface/DYNReactiveCurvePointsInterface.h:32]: a point of a PQ diagram extracted from network file

**Review by cossartque — CHANGES_REQUESTED** [review]

File: dynawo/sources/Modeler/DataInterface/DYNGeneratorInterface.h (hunk 2 above)
Comment (lecourtoisflo) [reply dynawo/sources/Modeler/DataInterface/DYNGeneratorInterface.h:133]: The previous API was named that way. To keep backwards compatibility, I kept the same name

File: dynawo/sources/Modeler/DataInterface/DYNReactiveCurvePointsInterface.h (hunk 3 above)
Comment (lecourtoisflo) [reply dynawo/sources/Modeler/DataInterface/DYNReactiveCurvePointsInterface.h:10]: OK. DFL doesn't use this line in its copyright

File: dynawo/sources/Modeler/DataInterface/DYNReactiveCurvePointsInterface.h (hunk 4 above)
Comment (lecourtoisflo) [reply dynawo/sources/Modeler/DataInterface/DYNReactiveCurvePointsInterface.h:32]: OK

File: dynawo/sources/Modeler/DataInterface/DYNGeneratorInterface.h (hunk 2 above)
Comment (cossartque) [reply dynawo/sources/Modeler/DataInterface/DYNGeneratorInterface.h:133]: Ok, thanks

File: dynawo/sources/Modeler/DataInterface/DYNReactiveCurvePointsInterface.h (hunk 4 above)
Comment (cossartque) [reply dynawo/sources/Modeler/DataInterface/DYNReactiveCurvePointsInterface.h:32]: Well if we talk about "reactive curve" or "reactive capability" it necessarily means that we are talking about a diagram

Comment (lecourtoisflo) [reply dynawo/sources/Modeler/DataInterface/DYNReactiveCurvePointsInterface.h:32]: OK

**Review by gautierbureau — APPROVED** [review]

**Comment by gautierbureau** [comment]
There is a conflict

**Pull request merged**